Sources/SwiftParser/Nominals.swift

```diff
@@ -339,6 +339,10 @@ extension Parser {
     )
   }
 
+  mutating func atInheritanceListTerminator() -> Bool {
+    return self.experimentalFeatures.contains(.trailingComma) && (self.at(.leftBrace) || self.at(.keyword(.where)))
+  }
+
```

Comment:
I think this doesn’t work in general because `associatedtype` can also have an inheritance clause and it is not followed by `{` or `where`. 

Eg. 

```swift
associatedtype Iterator: IteratorProtocol, 
```

would probably fail to parse wit this change.

I think we should limit the trailing comma to the types where it is guaranteed to be followed by `where` or `{` (ie. everything except `associatedtype`).

Reply:
@ahoppen I believe it's just a matter of update the proposal text mentioning that trailing comma won't work for inheritance clause of an associatedtype, right? The same for where clauses for protocol definition. In general, make it clear that trailing comma will only be allowed when there's a clear terminator.

Reply:
I’m only speaking as a reviewer of this PR here but I think it would make sense to exclude trailing commas for these situations without a clear terminator from the evolution proposal for now. 

There might be ways of getting them right, but doing so is more complicated than the cases that have a clear terminator. Highlighting these situations in a *Future directions* section would make sense, I think. That way your proposal can focus on the majority of cases and if we do choose to extend trailing commas eg. `case` lists, the corner cases will get the attention they deserve.

Reply:
I've updated the proposal text highlighting these cases. Thanks for catching them.